### ISSUE TITLE
 fzf shell integration failing

### ISSUE THREAD

**Issue by Nicknamely** [ohmyzsh/ohmyzsh#12629]
### Describe the bug

fzf does not recognize `--zsh` option.

### Steps to reproduce

1.  Add  `eval "$(fzf --zsh)"` to `.zshrc` file
2. save then quit
3. run `zsh ` or open new terminal or run `source ~/.zshrc`

### Expected behavior

```
`[WARNING]: Console output during zsh initialization detected.

When using Powerlevel10k with instant prompt, console output during zsh
initialization may indicate issues.

You can:

  - Recommended: Change ~/.zshrc so that it does not perform console I/O
    after the instant prompt preamble. See the link below for details.

    * You will not see this error message again.
    * Zsh will start quickly and prompt will update smoothly.

  - Suppress this warning either by running p10k configure or by manually
    defining the following parameter:

      typeset -g POWERLEVEL9K_INSTANT_PROMPT=quiet

    * You will not see this error message again.
    * Zsh will start quickly but prompt will jump down after initialization.

  - Disable instant prompt either by running p10k configure or by manually
    defining the following parameter:

      typeset -g POWERLEVEL9K_INSTANT_PROMPT=off

    * You will not see this error message again.
    * Zsh will start slowly.

  - Do nothing.

    * You will see this error message every time you start zsh.
    * Zsh will start quickly but prompt will jump down after initialization.

For details, see:
https://github.com/romkatv/powerlevel10k#instant-prompt

-- console output produced during zsh initialization follows --`
```

### Screenshots and recordings

In my `~/.zshrc` file
![image](https://github.com/user-attachments/assets/1125ee3e-ccec-425b-bb9e-469d8f72f53b)

After running `zsh` or opening new terminal
![image](https://github.com/user-attachments/assets/50f465f8-204f-4feb-824e-c94268344a76)

### OS / Linux distribution

windows 10

### Zsh version

5.8

### Terminal emulator

windows terminal

### If using WSL on Windows, which version of WSL

WSL2

### Additional context

`fzf --version` : 
 `0.29 (devel)  `

Ubuntu (wsl2)

**Comment by carlosala** [comment]
Could you share the output of `setopt`?

**Comment by Nicknamely** [comment]
```❯ setopt
histfindnodups
histignorealldups
histignoredups
histignorespace
histsavenodups
interactive
monitor
promptsubst
sharehistory
shinstdin
zle
```

**Comment by carlosala** [comment]
Oh yeah now I see it. I thought you were using `fzf` ohmyzsh plugin. `fzf --zsh` is only available from `fzf` 0.48 ahead. You are running `fzf` 0.29.
That's not an ohmyzsh problem. Thanks for the report!
